src/libs/antares/study/parts/thermal/cluster_list.cpp

```diff
@@ -721,7 +721,7 @@ void ThermalClusterList::ensureDataPrepro()
     {
         auto c = it->second;
         if (not c->prepro)
-            c->prepro = new PreproThermal(c.get());
+            c->prepro = new PreproThermal(c);
```

Comment:
Why not : 
```c++
c->prepro = new PreproThermal(c->shared_from_this());
```
?

Reply:
Because `c` is already a `shared_ptr`, there is no need to create a new one.

Reply:
ok, I'll approve then